Makefile

```diff
@@ -41,6 +44,8 @@ help:
 	@echo
 	@echo 'Configuration options:'
 	@echo
+	@echo '  BUILD_TYPE              Is it a release build or a development build? [default=development]'
+	@echo '  BUILD_ARGUMENTS         Space separated list of build arguments. [e.g. "COMPUTE_BACKENDS=htcondorcern" no build arguments are passed by default]'
```

Comment:
As discussed IRL a few days ago, aren't we going to separate arguments into their own respective environment variables?

Instead of:

```console
$ BUILD_ARGUMENTS="COMPUTE_BACKENDS=htcondorcern,slurmcern VERBOSITY_LEVEL=info" make foo
```

going for separating them:

```
$ BUILD_COMPUTE_BACKENDS=htcondorcern,slurmcern BUILD_VERBOSITY_LEVEL=info make ci
```

This is the usual envvar usage paradigm, and it could also help us to distinguish target components one day, for example:

```console
$ export REANA_JOB_CONTROLLER_COMPUTE_BACKENDS=htcondorcern,slurmcern
$ export REANA_SERVER_VERBOSITY=info
$ make ci
```

Just some examples.

P.S. We can do this also after merge.